### PR TITLE
Added Non-Blind Mode Options

### PR DESCRIPTION
**Pull request by zrhernan**
The following updates were made to the repository:
 - Decision message box at the start to indicate whether to view allocator in 'blind' or 'non-blind' mode
 - If in 'non-blind' mode, the following features were added for the user's convenience when adding a new subject entry:
   - Dialog message box showing results of the Frane algorithm which includes: final group placement for the new entry as well as group-wise and group-field-wise probability values
   - Second decision message box asking whether the user wishes to keep the current group placement or switch new subject entry to the opposite group
   - Third dialog message box confirming the new entry group placement
 - In either blind/non-blind mode, a new feature was added to code such that 'BCI group' subject IDs matched to a 'Sham group' subject are saved and compared so that a 'BCI group' subject can be paired to ONLY ONE 'Sham group' subject
 - If in 'blind' mode, the allocator operates as originally designed, with automatically group placement of a new subject entry that is blind to the user inputting subject data

### CODE REVIEW

File: patientalloc/src/Database/Database.py [hunk 1]
```diff
@@ -196,6 +201,7 @@ def getPValue(self, field):
         groups = ({self.groups[0]: [], self.groups[1]: []})
         pvalue = 0
         entry_number = 0
+        # print(self.entries)
```
Comment (Ruijan) [inline patientalloc/src/Database/Database.py]: Remove commented out code

File: patientalloc/src/Database/Subject/SubjectFactory.py [hunk 2]
```diff
@@ -3,24 +3,123 @@
 
 
 class SubjectFactory:
-    def __init__(self, database, settings):
+    def __init__(self, database, settings, currentGui):
+        self.gui = currentGui
```
Comment (Ruijan) [inline patientalloc/src/Database/Subject/SubjectFactory.py]: The subject factory should not know about the GUI. It is independent of the way the database is displayed. Move the display you added to the DatabaseLoaderDisplay.py in GUI and implement it in the add_subject method.

File: patientalloc/src/GUI/GUI.py [hunk 3]
```diff
@@ -45,6 +45,12 @@ def __init__(self, mode):
             self.currentFrame = DatabaseLoaderDisplay(self)
             self.currentFrame.display()
 
+        viewmode = self.getViewMode()
```
Comment (Ruijan) [inline patientalloc/src/GUI/GUI.py:48]: There is an admin version of the GUI. The idea was to not forbid users to see the results except those who can run it in admin mode. It's the first argument passed to the GUI creation in patientalloc/__init__.py.

File: build/lib.linux-x86_64-2.7/patientalloc/src/Database/Subject/SubjectFactory.py [hunk 4]
```diff
@@ -1,27 +1,109 @@
 import patientalloc
```
Comment (Ruijan) [inline build/lib.linux-x86_64-2.7/patientalloc/src/Database/Subject/SubjectFactory.py:1]: It seems that the whole build folder has been added to the commit.
I forgot to add the build folder to the .gitignore. Could you add it?
Thanks

**Review by Ruijan — CHANGES_REQUESTED** [review]
Nice feature but please be careful to keep the MVC pattern intact.
The Subject and SubjectFactory must not know anything about the GUI.

File: patientalloc/src/Database/Subject/SubjectFactory.py (hunk 2 above)
Comment (zrhernan) [reply patientalloc/src/Database/Subject/SubjectFactory.py]: I implemented these changes such that all GUI definitions are called from the GUI.py script in the add_subject method of the DatabaseLoaderDisplay.py script.

File: build/lib.linux-x86_64-2.7/patientalloc/src/Database/Subject/SubjectFactory.py (hunk 4 above)
Comment (zrhernan) [reply build/lib.linux-x86_64-2.7/patientalloc/src/Database/Subject/SubjectFactory.py:1]: Yes I have added 'build' to the .gitignore file.

File: patientalloc/src/Database/Database.py (hunk 1 above)
Comment (zrhernan) [reply patientalloc/src/Database/Database.py]: Will do, I'll remove any commented code remaining in the files.

File: patientalloc/src/GUI/GUI.py (hunk 3 above)
Comment (zrhernan) [reply patientalloc/src/GUI/GUI.py:48]: I understand this, however this was a feature Jose specifically asked to add to the code. Where an option is given at the start of the allocation program whether to view all relevant information related to group placement of the new subject entry (probabilities when placed in either group for all fields/by each field and option to switch groups) or not view any information at all so as to keep the subject blind to the group placement. My understanding is that you can only create a database in the admin mode. If you have any suggestions on this or anything to correct that I said, please feel free to message back.

**Review by Ruijan — APPROVED** [review]
All good to me. Thank you for taking my comments into account.
Are you going to deploy this code to the Beau-séjour clinic? If yes, you will have to rename a lot of files because of your change from "reaching" to "extension".
Don't hesitate to contact me directly if you need: rechenmann@gmail.com

File: patientalloc/src/Database/Subject/SubjectFactory.py (hunk 2 above)
Comment (Ruijan) [reply patientalloc/src/Database/Subject/SubjectFactory.py]: Perfect. Thanks!

File: patientalloc/src/GUI/GUI.py (hunk 3 above)
Comment (Ruijan) [reply patientalloc/src/GUI/GUI.py:48]: If the requirements changed then it's fine.

File: patientalloc/src/GUI/GUISettings.py [hunk 5]
```diff
@@ -12,16 +12,16 @@
 
 class GUISettings():
     def __init__(self):
-        xml_file_path = str(Path.home()) + "/.cnbitk/cnbimi/xml/"
+        xml_file_path = str(Path.home()) + "/StrokeMI/protocolxmls/"
```
Comment (zrhernan) [inline patientalloc/src/GUI/GUISettings.py]: I'll need to change these back to their original filepaths

**Review by zrhernan — COMMENTED** [review]
I believe I changed changed the 'reaching' string back to 'extension'. In addition I also reverted the directory paths in GUISettings back to their original paths. They should all be in the latest commit.

**Comment by zrhernan** [comment]
I think this still needs to be merged. It doesn't seem to have ever been merged yet.

**Pull request closed without merging**